### PR TITLE
Cleaned up React Context

### PR DESCRIPTION
**Pull request by annoyingpr**
## Briefly describe the feature or bug that this PR addresses.

Cleaned up some of the unused variables in React Context. Data page state variables must be added when filters are added to that page. 

## Please link the Airtable ticket associated with this PR.
https://airtable.com/tbli2lWQHAqBa6ZcI/viw8Ro4zYPYcBGXRw/recjrEU9Ba07eYuEe?blocks=hide

## If applicable, include screenshots of the feature/bugfix introduced by this PR (Please include both desktop and mobile if there is a significant UI change).
No UI changes, everything should work normally. 

## Describe the steps you took to test the feature/bugfix introduced by this PR.
Just tested around the website locally to see if any functionality was broken.

## Does this PR depend on any recent backend work? If so, please include the PR number from the iit-backend repo and associated Airtable ticket link.
Nope.

### CODE REVIEW

File: src/context.tsx [hunk 1]
```diff
@@ -41,147 +50,124 @@ export function getDefaultFilters(): Filters {
...
-  },
   allFilterOptions: getEmptyFilters(),
   dataPageState: {
     exploreIsOpen: true,
-    showStudiesModal: false,
-    routingOccurred: false
   },
   calendarStartDates: {
     // Important, the fact that we use an hour here tells us that we are using a default value
     minDate: new Date(2019, 1, 1, 1),
```
Comment (annoyingpr) [inline src/context.tsx:78]: Is it a bad idea to use the `minDate` and `maxDate` from context for `publish_date` in these two functions:

**getEmptyFilters()**
image

**getDefaultFilters()**
image

Comment (atatmaja) [reply src/context.tsx:78]: I'm not opposed, makes it a bit more readable

File: src/context.tsx [hunk 2]
```diff
@@ -16,23 +25,23 @@ export function getEmptyFilters(): Filters {
     isotypes_reported: new Set(),
     specimen_type: new Set(),
     publish_date: new Set([new Date(2019, 1, 1, 1), new Date()]),
-    estimate_grade: new Set()
-  }
+    estimate_grade: new Set(),
+  };
 }
 
 export function getDefaultFilters(): Filters {
```
Comment (atatmaja) [inline src/context.tsx:32]: might need to double check but I don't think we need getDefaultFilters anymore as this was for the analyze tab

File: src/context.tsx [hunk 3]
```diff
@@ -190,22 +176,27 @@ const reducer = (state: State, action: Record<string, any>): State => {
       const pageState = newState[pageStateEnum as keyof State] as PageState;
       pageState.isLoading = isLoading;
       return newState;
-    };
+    }
     case "UPDATE_AGGREGATION_FACTOR":
       return { ...state, chartAggregationFactor: action.payload };
     case "UPDATE_EXPLORE_IS_OPEN":
```
Comment (atatmaja) [inline src/context.tsx:182]: Could be wise to address in another PR but this whole system around our `dataPageState` object is very outdated. We should update this logic to be simpler and tie up loose ends eventually. Maybe we have a quick sync about this sometime @brett-stephen @annoyingpr

**Review by atatmaja — COMMENTED** [review]
A few comments, thanks for doing this clean up!

File: src/context.tsx (hunk 3 above)
Comment (brett-stephen) [reply src/context.tsx:182]: Agreed. Also, a bit unrelated but it would also be nice to address the issue of the API calls saved in the app context. If you recall, we didn't have a way to distinguish between a failed API call and one that hasn't returned anything yet.

**Comment by brett-stephen** [comment]
Once Austins concern has been addressed it should be good to merge. Looks good!

**Review by brett-stephen — COMMENTED** [review]
Once Austins concern has been addressed it should be good to merge. Looks good!

File: src/context.tsx (hunk 3 above)
Comment (annoyingpr) [reply src/context.tsx:182]: Sounds good, do you think it should be a part of this [ticket](https://airtable.com/tbli2lWQHAqBa6ZcI/viw8Ro4zYPYcBGXRw/recNqmd8naERYKBio?blocks=hide) or should we create a new one?

File: src/context.tsx (hunk 2 above)
Comment (annoyingpr) [reply src/context.tsx:32]: Great catch!

File: src/context.tsx [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
+  PageState,
+  State,
+} from "./types";
 
-export const AppContext = createContext({} as [State, Dispatch<Record<string, any>>]);
+export const AppContext = createContext(
+  {} as [State, Dispatch<Record<string, any>>]
+);
+
+const minDate = new Date(2019, 1, 1, 1);
```
Comment (annoyingpr) [inline src/context.tsx:15]: Lemme know if you guys don't like this approach - I had to do something like this because of the circular dependencies between `initialState` and `getEmptyFilters`.

File: src/context.tsx (hunk 3 above)
Comment (atatmaja) [reply src/context.tsx:157]: make new one pls

File: src/context.tsx [hunk 5]
```diff
@@ -15,173 +26,123 @@ export function getEmptyFilters(): Filters {
...
       };
     case "UPDATE_FILTER": {
       const { pageStateEnum, filterType, filterValue } = action.payload;
       const newState = { ...state };
       const pageState = newState[pageStateEnum as keyof State] as PageState;
       pageState.filters[filterType as FilterType] = new Set(filterValue);
       return newState;
-    };
+    }
     // Note: deprecated but leaving here in case we ever need again
```
Comment (atatmaja) [inline src/context.tsx:132]: on second thought what do we think about just deleting this lol. i think I made a bad call leaving here and putting a deprecated comment

File: src/context.tsx (hunk 4 above)
Comment (atatmaja) [reply src/context.tsx:15]: eh readable enough for now

Comment (atatmaja) [inline src/context.tsx:15]: Let's make a small ticket to make this min date reflective of our database state (earliest date in our database). Not a huge priority imo but would be nice @annoyingpr

**Review by atatmaja — APPROVED** [review]
Looks good as long as we've tested thoroughly.

Few action items:
- Pls make tickets that I suggested
- Delete the UPDATE_META_ANALYSIS reducer function and do a quick smoke test to make sure nothing breaks

Thanks for the work!

File: src/context.tsx (hunk 4 above)
Comment (atatmaja) [reply src/context.tsx:15]: Also nit: what do we think about renaming these consts to `initialMaxDate` and `initialMinDate` to make sure these are the initial bounds on the calendar

File: src/context.tsx (hunk 5 above)
Comment (annoyingpr) [reply src/context.tsx:132]: like it was never there!
